src/azure-cli/azure/cli/command_modules/acr/connected_registry.py

```diff
@@ -0,0 +1,494 @@
+            sync_properties=SyncProperties(
+                token_id=sync_token_id,
+                schedule=sync_schedule,
+                message_ttl=sync_message_ttl,
+                sync_window=sync_window
+            )
+        ),
+        client_token_ids=client_token_list,
+        logging=LoggingProperties(
+            log_level=log_level,
+            audit_log_status='Enabled' if sync_audit_logs_enabled else 'Disabled'
+        )
+    )
+
+    try:
+        return client.create(subscription_id=subscription_id,
+                             resource_group_name=resource_group_name,
+                             registry_name=registry_name,
+                             connected_registry_name=connected_registry_name,
+                             connected_registry_create_parameters=connected_registry_create_parameters)
+    except ValidationError as e:
+        raise CLIError(e)
+
+
+def acr_connected_registry_update(cmd, # pylint: disable=too-many-locals, too-many-statements
+                                  client,
+                                  registry_name,
+                                  connected_registry_name,
+                                  add_client_token_list=None,
+                                  remove_client_token_list=None,
+                                  resource_group_name=None,
+                                  sync_schedule=None,
+                                  sync_window=None,
+                                  log_level=None,
+                                  sync_message_ttl=None,
+                                  sync_audit_logs_enabled=None):
+    _, resource_group_name = validate_managed_registry(
+        cmd, registry_name, resource_group_name)
+    subscription_id = get_subscription_id(cmd.cli_ctx)
+    current_connected_registry = acr_connected_registry_show(
+        cmd, client, connected_registry_name, registry_name, resource_group_name)
+
+    # Add or remove from the current client token id list
+    if add_client_token_list is not None:
+        for i, client_token_name in enumerate(add_client_token_list):
+            add_client_token_list[i] = build_token_id(
+                subscription_id, resource_group_name, registry_name, client_token_name)
+        add_client_token_set = set(add_client_token_list)
+    else:
+        add_client_token_set = set()
+    if remove_client_token_list is not None:
+        for i, client_token_name in enumerate(remove_client_token_list):
+            remove_client_token_list[i] = build_token_id(
+                subscription_id, resource_group_name, registry_name, client_token_name)
+        remove_client_token_set = set(remove_client_token_list)
+    else:
+        remove_client_token_set = set()
+
+    duplicate_client_token = set.intersection(add_client_token_set, remove_client_token_set)
+    if duplicate_client_token:
+        errors = sorted(map(lambda action: action[action.find('/') + 1:], duplicate_client_token))
```

Comment:
what is this logic?

Reply:
good catch! this should be rfind. The idea is to display only the token name. No the whole ID and the +1 is to start one character after the '/' match